### ISSUE TITLE
vcpkg crashed, mistyped upgrade no dry run command,  user at fault.

### ISSUE THREAD

**Issue by GeorgePimpleton** [microsoft/vcpkg#35847]
I typed what I thought was my usual vcpkg upgrade command and received the following message:

D:\Programming\vcpkg>vcpkg upgrade -no-dry-run >> install.txt
internal error: D:\a\_work\1\s\src\vcpkg\commands.upgrade.cpp(179): vcpkg has crashed; no additional details are available.
Please open an issue at https://github.com/microsoft/vcpkg/issues/new?template=other-type-of-bug-report.md&labels=category:vcpkg-bug with detailed steps to reproduce the problem.

While I realize the issue is solely my cause by my mistyping mistake maybe vcpkg needs to allow for simple mistakes and terminate with more "grace" and user-friendly error messages.

Red-flagging, vcpkg crashing from a very human typing mistake made it difficult for me to decipher right away what I had done wrong.

**Comment by autoantwort** [comment]
1. `git pull`
2. `./bootstrap.bat`
3. Try again

**Comment by Cheney-W** [comment]
Yes, in the latest vcpkg, when encountering input errors like this, the following prompt will be provided:
![image](https://github.com/microsoft/vcpkg/assets/38240633/2b06c489-6912-44ef-9f98-91bf8a6ee25f)
